### PR TITLE
Add support to signals and slots

### PR DESCRIPTION
**Pull request by mirko-bianco**

### CODE REVIEW

File: source/Slots/Fido.Slots.pas [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+      FMessage: string;
+      FSlot: Spring.TAction<TArray<TValue>>;
+      FSlotType: TSlotType;
+    public
+      constructor Create(const SignalActor: IObservable; const Message: string; const SlotType: TSlotType; const Slot: Spring.TAction<TArray<TValue>>);
+
+      function SignalActor: IObservable;
+      function Message: string;
+      function SlotType: TSlotType;
+      function Slot: Spring.TAction<TArray<TValue>>;
```
Comment (BasvRossem) [inline source/Slots/Fido.Slots.pas:58]: Why not use a read only property? This seems a bit redundant...

**Review by BasvRossem — CHANGES_REQUESTED** [review]
Could you also create some tests for your slots?

**Comment by mirko-bianco** [comment]
> Could you also create some tests for your slots?

I've been a sloth...

**Comment by mirko-bianco** [comment]
> Could you also create some tests for your slots?


done

**Review by BasvRossem — APPROVED** [review]

**Pull request merged**